virotour/lib/src/tour/tour_list_view.dart

```diff
@@ -1,70 +1,134 @@
+import 'dart:convert';
+
 import 'package:flutter/material.dart';
+import 'package:http/http.dart' as http;
 import 'package:virotour/src/settings/settings_view.dart';
 import 'package:virotour/src/tour/tour.dart';
 import 'package:virotour/src/tour/tour_details_view.dart';
 
-class TourListView extends StatelessWidget {
+class TourListView extends StatefulWidget {
   const TourListView({
-    super.key,
-    // TODO: Get the list of tours from GET /
-    this.items = const [Tour(1), Tour(2), Tour(3)],
-  });
+    Key? key,
+    required this.items,
+  }) : super(key: key);
 
   static const routeName = '/';
 
   final List<Tour> items;
 
+  @override
+  _TourListViewState createState() => _TourListViewState();
+}
+
+class _TourListViewState extends State<TourListView> {
+  late Future<List<Tour>> _tourData;
+
+  @override
+  void initState() {
+    super.initState();
+    _tourData = fetchData();
+  }
+
+  static Future<List<Tour>> fetchData() async {
+    // TODO: Implement fetching the tour data from an API endpoint
+    // final response = await http.get(Uri.parse('https://virotour.com/tours/endpoint'));
+    final response = [
+      {
+        'data': {
+          "tours": [
+            {
+              "tour_name": "example tour 1",
+              "description": "This is an example tour",
+              "id": 1
+            },
+            {
+              "tour_name": "example tour 2",
+              "description": "This is another example tour",
+              "id": 2
+            }
+          ]
+        }
+      }
+    ];
+
+    // if (response.statusCode == 200) {
+    //   final responseData = jsonDecode(response.body);
+    final data = response[0]['data']!['tours'] as List<dynamic>;
+
+    return data
+        .map((tour) => Tour(
+              id: tour['id'].toString(),
+              tourName: tour['tour_name'].toString(),
+              description: tour['description'].toString(),
+            ))
+        .toList();
+    // } else {
+    //   throw Exception("Failed to load tour data!");
+    // }
+  }
+
   @override
   Widget build(BuildContext context) {
     return Scaffold(
       appBar: AppBar(
         title: const Text('Tours'),
         actions: [
+          IconButton(
+            icon: const Icon(Icons.refresh),
```

Comment:
```suggestion
            icon: const Icon(Icons.refresh),
```
We shouldn't need to include the refresh button for 2 reasons:
1. It doesn't match our design
2. The users once open the app, the app will automatically fetch the list of tours. Once the users create a new tour, after saving the new tour, the app will fetch the list of tours.

We should make the fetch() triggered on page loaded.

Reply:
My thinking was that because we don't have user authentication and all tours are visible to all users, if one user created a tour when someone else was in app, they could refresh and see them.  If we don't have a refresh button, the user would have to either restart the app or create a new tour to get a refreshed tours list. 

We could always add a refresh icon to our design images.  Maybe we should have a discussion about how usable we need to make the app for the "complete" status of our project.

Reply:
That's a reasonable suggestion. If you feel strongly about adding another button to the menu, let's discuss it with the group.

To address your concerns:
- if one user created a tour when someone else was in app, they could refresh and see them -> This can be done by refreshing the webpage or pulling the app screen down on mobile devices. They created the gesture for this exact reason so users won't have to restart the app.

Reply:
You can also use this: https://api.flutter.dev/flutter/material/RefreshIndicator-class.html